golang/src/config.rs

```diff
@@ -0,0 +1,238 @@
+use libc::c_char;
+use pravega_client_config::credentials::Credentials;
+use pravega_client_config::*;
+use pravega_client_retry::retry_policy::RetryWithBackoff;
+use pravega_client_shared::*;
+use std::ffi::CStr;
+use std::ops::Add;
+use std::time::{Duration, Instant};
+
+const SPLIT: &str = ",";
+
+#[repr(C)]
+pub struct StreamConfigurationMapping {
+    pub scope: *const c_char,
+    pub stream: *const c_char,
+    pub scaling: ScalingMapping,
+    pub retention: RetentionMapping,
+    pub tags: *const c_char, // split by ','
+}
+
+impl StreamConfigurationMapping {
+    pub fn to_stream_configuration(&self) -> StreamConfiguration {
+        let scope = to_str(self.scope);
+        let stream = to_str(self.stream);
+        let scope_stream = scope.to_string() + "/" + stream;
+        StreamConfiguration {
+            scoped_stream: ScopedStream::from(scope_stream.as_str()),
+            scaling: self.scaling.to_scaling(),
+            retention: self.retention.to_retention(),
+            tags: str_to_tags(self.tags),
+        }
+    }
+}
+
+fn to_str<'a>(p_str: *const c_char) -> &'a str {
+    let raw = unsafe { CStr::from_ptr(p_str) };
+    raw.to_str()
+        .expect("Error: Failed to convert C string to Rust string.")
```

Comment:
The most probable reason is that the passed value is not valid utf-8. This should be mentioned in the error message.